docs/architecture/index.rst

```diff
@@ -1,25 +1,586 @@
+.. _det-system-architecture:
+
+.. _system-architecture:
+
 #####################
  System Architecture
 #####################
 
-Determined consists of a single **master** and one or more **agents**. There is typically one agent
-per compute server; a single machine can serve as both a master and an agent.
+.. pull-quote::
+
+   Determined makes it easy to transform data into probabilities.
+
+   -- Garrett Goon
+
+**********
+ Overview
+**********
+
+This document describes the Determined system architecture in terms of its components and behavior,
+providing context for both system administrators and machine learning engineers before exploring
+Determined in more detail.
+
+Determined implements the training workflow shown in the **Run Training** section of the following
+figure:
+
+.. image:: /assets/images/arch12.png
+
+**TBD: Describe what the relevant boxes are and how they interact; e.g., built-in support for data,
+checkpointing, and metric storage.**
+
+The following figure is a more detailed view of the Determined platform:
+
+.. image:: /assets/images/arch11.png
+
+**TBD: Describe what the boxes are and how they interact. And, of course, generalize the diagram.**
+
+************
+ Components
+************
+
+Determined deployment can range from a local laptop and on-premises setups to distributed master and
+agent nodes, all with database and file storage connectivity, containerization, and cloud resource
+options.
+
+The following figure shows the key master and agent instances on separate machines:
+
+.. image:: /assets/images/arch00.png
+
+However, a single machine can have both a master and one agent instance.
+
+Master and Agent Nodes
+======================
+
+Determined comprises a single *master* and one or more *agents* in a cluster environment. The master
+is a single, non-GPU node that manages the cluster.
+
+The master keeps experiment metadata in a PostgreSQL database, which can be queried using the WebUI
+or CLI. You can interact with the master, directly, but cannot interact with agents.
```

Comment:
> You can interact with the master, directly, but cannot interact with agents.

I am not sure what the goal of this statement is.  It seems to be maybe correct, based on the interpretation of "interact".  For example, you can disable agents (via the master), that's sort of interacting.

Maybe you mean something more like: "Most user interaction involves the WebUI or CLI communicating with the master" or something?